reframe/core/schedulers/torque.py

```diff
@@ -35,6 +35,14 @@
 class TorqueJobScheduler(PbsJobScheduler):
     TASKS_OPT = '-l nodes={num_nodes}:ppn={num_cpus_per_node}'
 
+    def _get_nodelist(self, job, stdout):
```

Comment:
For symmetry with the Slurm backend rename this as follows:

```suggestion
    def _set_nodelist(self, job, nodespec):
```

Also you should do the `re.search()` outside the function and check here if the nodelist is already set. Have a look at the corresponding function in the Slurm backend.